### ISSUE TITLE
Getting error while running Terraform plan - provider.docker: Error initializing Docker client: protocol not available

### ISSUE THREAD

**Issue by jay8127** [terraform-providers/terraform-provider-docker#205]
This is my main.tf, which refers to two different modules. 
```hcl
module "AzureVM" {
    source = "./AzureVM"
    }

module "Docker" {
    source = "./Docker"
    
    vm_ip_address = "${module.AzureVM.public_ip_address}"
}

And this is my docker provider in "Docker" module.
provider "docker" {
    host = "tcp://${var.vm_ip_address}:2376"
    alias = "docker"
}
```

My design is that set up a VM on Azure using Azure provider, and use the IP address dynamically into Docker Provider. 
Terraform init works. And I understand that I have to run Terraform apply separately for each module. But running Terraform plan fails with below error.
```console
Error: Error running plan: 1 error(s) occurred:
* provider.docker: Error initializing Docker client: protocol not available
```

I am running these commands in Windows machine, but Azure VM is Ubuntu. So I thought to switch docker desktop to Linux, which I did but still same issue. And I tried to find answer but no luck yet. Can you please help me what could be the issue?

**Comment by dubo-dubon-duponey** [comment]
Hey @jay8127 

Can you verify that:

 * docker is configured to expose a tcp socket (this is not the default behavior IIRC)
 * verify that you can reach that socket on {AzureVM.public_ip_address}:2376 from your local machine (without terraform)

That being said, exposing a docker socket publicly on the internet (assuming you didn't restrict access to it otherwise), is probably not a good idea.

I would highly recommend using ssh instead...

**Comment by mavogel** [comment]
@jay8127 could you verify the 2 points asked in the comment above?

**Comment by jay8127** [comment]
@mavogel Sorry for late response, I verified it and it works. But I noted the other comment you made regarding security, "That being said, exposing a docker socket publicly on the internet (assuming you didn't restrict access to it otherwise), is probably not a good idea.", which makes sense. So I ended up changing design. Now I am not using Docker Provider. I mean, I dont want to execute docker commands from any machine exposing docker socket publicly. So I started using K8s provider instead and set up container there. It fulfills my requirement. This may not be possible for all use cases but it suits my use case.
